### PR TITLE
feat: updated tests for Liquidity and Exchange contracts

### PR DESCRIPTION
**Pull request by RauppRafael**

### CODE REVIEW

File: src/helpers/Fees.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        )
+        const liquidityPercentage = liquidityInvestments.reduce(
+            (acc, investment) => acc + investment.percentage,
+            BigInt(0),
+        )
+        const tokensPercentage = tokenInvestments.reduce(
+            (acc, investment) => acc + investment.percentage,
+            BigInt(0),
+        )
+
```
Comment (matheus-pissaia) [inline src/helpers/Fees.ts:44]: Perhaps we could have a function that calculates all of this, something like:
```ts
function calculateInvestmentPercentages(strategyProducts) {
    return strategyProducts.map(productInvestments => 
        productInvestments.reduce((acc, curr) => acc + curr.percentage, BigInt(0))
    )
}

const [
    dcaPercentage,
    vaultPercentage,
    liquidityPercentage
] = calculateInvestmentPercentages([
    dcaInvestments,
    vaultInvestments,
    liquidityInvestments,
])
```

File: src/helpers/Fees.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        subscribedUser: boolean,
+        dca: UseFee,
+        vaultManager: UseFee,
+        liquidityManager: UseFee,
+        exchangeManager: UseFee,
+    ) {
+        const {
+            protocolFee,
+            strategistFee,
+        } = await this.getStrategyFeePercentage(
```
Comment (matheus-pissaia) [inline src/helpers/Fees.ts:106]: Avoid unexpected issues with:
```suggestion
        } = await Fees.getStrategyFeePercentage(
```

File: test/LiquidityManager/invest.test.ts [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+            )
+    }
+
+    function validateInvestTransaction(
+        receipt: ContractTransactionReceipt | null,
+        amount0: bigint,
+        token0Price: BigNumber,
+        amount1: bigint,
+        token1Price: BigNumber,
+    ) {
```
Comment (RauppRafael) [inline test/LiquidityManager/invest.test.ts:173]:
```suggestion
        amount1: bigint,
        price0: BigNumber,
        price1: BigNumber,
    ) {
```

what do you think?

File: test/LiquidityManager/invest.test.ts [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+
+        Compare.almostEqualPercentage({
+            target: amount,
+            value: BigInt(
+                new BigNumber(amount0.toString()).times(token0Price)
+                    .plus(new BigNumber(amount1.toString()).times(token1Price))
+                    .toString(),
+            ),
+            tolerance: new BigNumber('0.1'),
+        })
```
Comment (RauppRafael) [inline test/LiquidityManager/invest.test.ts:201]: 10% tolerance is too much, i think it should be about 1-2%

File: test/LiquidityManager/invest.test.ts (hunk 3 above)
Comment (matheus-pissaia) [reply test/LiquidityManager/invest.test.ts:173]: agree

File: test/LiquidityManager/invest.test.ts (hunk 4 above)
Comment (matheus-pissaia) [reply test/LiquidityManager/invest.test.ts:201]: Oh its 10, i thought it was like 0.1% 🤣

**Pull request merged**